### PR TITLE
fix: Update channel pin timestamp

### PR DESCRIPTION
**Pull request by leestarb**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition

## Description

fixes error when updating timestamp to `None`:

```
TypeError: Timestamp must be one of: datetime, int, float, ISO8601 str, got `None`
```

## Changes

- Remove redundant convert

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/api/events/processors/channel_events.py [hunk 1]
```diff
@@ -40,7 +39,7 @@ async def _on_raw_channel_update(self, event: "RawGatewayEvent") -> None:
     @Processor.define()
     async def _on_raw_channel_pins_update(self, event: "RawGatewayEvent") -> None:
         channel = await self.cache.get_channel(event.data.get("channel_id"))
-        channel.last_pin_timestamp = timestamp_converter(event.data.get("last_pin_timestamp"))
+        channel.last_pin_timestamp = event.data.get("last_pin_timestamp")
```
Comment (silasary) [inline dis_snek/api/events/processors/channel_events.py:42]: My gut says you're changing this from a `datetime` to a `str`

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]:
```suggestion
        if last_pin_timestamp := event.data.get("last_pin_timestamp"):
            channel.last_pin_timestamp = timestamp_converter(last_pin_timestamp)
```

I think something like this make more sense

Comment (leestarb) [reply dis_snek/api/events/processors/channel_events.py:42]: I think it's ok because attrs will convert this automatically

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]: The last time i read attrs docs, the converter only happens on init, not property set. Have you tested it?

Comment (leestarb) [reply dis_snek/api/events/processors/channel_events.py:42]: > The last time i read attrs docs, the converter only happens on init, not property set. Have you tested it?

Yea, I'll try again

Comment (leestarb) [reply dis_snek/api/events/processors/channel_events.py:42]: Converter works on set!

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]: that is odd . I suspect the json parsing converts datetime for you.

image

attrs docs itself quotes...
> Please note that converters only run on initialization.

Comment (leestarb) [reply dis_snek/api/events/processors/channel_events.py:42]: > that is odd . I suspect the json parsing converts datetime for you.
> 
> image
> 
> attrs docs itself quotes...
> 
> > Please note that converters only run on initialization.

Test it yourself!

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]: image
sure

Comment (leestarb) [reply dis_snek/api/events/processors/channel_events.py:42]: > image sure

I was testing this on real update pin timestamp in dis_snek, try it, please

Comment (artem30801) [reply dis_snek/api/events/processors/channel_events.py:42]: Guys, don't fight
Attr runs converters and validators only on `init` *by default*
You can also enable this behavior on `setattr` too when you define your attr model
[Here](https://github.com/Discord-Snake-Pit/Dis-Snek/blob/master/dis_snek/client/utils/attr_utils.py#L22) we enable this behavior for all of our models defined using our `@attr_utils.define` decorator

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]: ah i recall what we did now lol... mb


image

we added this

Comment (benwoo1110) [reply dis_snek/api/events/processors/channel_events.py:42]: remind me to not review code when I'm halfway through a school report haha

**Review by benwoo1110 — APPROVED** [review]

**Review by artem30801 — APPROVED** [review]

**Review by silasary — APPROVED** [review]

**Pull request merged**